ios/keyman/Keyman/Keyman/MainViewController.swift

```diff
@@ -0,0 +1,1200 @@
+          let oskFont = kbDict[kKeymanOskFontKey]!
+          KMManager.sharedInstance().addKeyboard(withID: kbID, languageID: langID, keyboardName: kbName,
+              languageName: langName, isRTL: isRTL, isCustom: false, font: font, oskFont: oskFont)
+          KMManager.sharedInstance().setKeyboardWithID(kbID, languageID: langID, keyboardName: kbName,
+              languageName: langName, font: font, oskFont: oskFont)
+        }
+      } else {
+        let kbName = info[kKeymanKeyboardNameKey] as! String
+        let langName = info[kKeymanLanguageNameKey] as! String
+        let isRTL = ((info[kKeymanKeyboardRTLKey] as! String) == "Y") ? true : false
+        let font = info[kKeymanFontKey] as! String
+        let oskFont = info[kKeymanOskFontKey] as! String
+        KMManager.sharedInstance().addKeyboard(withID: nil, languageID: nil, keyboardName: kbName,
+            languageName: langName, isRTL: isRTL, isCustom: false, font: font, oskFont: oskFont)
+        KMManager.sharedInstance().setKeyboardWithID(nil, languageID: nil, keyboardName: kbName,
+            languageName: langName, font: font, oskFont: oskFont)
+      }
+    } else if let infoArray = kbInfo as? [[String: Any]] {
+      for info in infoArray {
+        let kbID = info[kKeymanKeyboardIdKey] as! String
+        let langID = info[kKeymanLanguageIdKey] as! String
+        let kbName = info[kKeymanKeyboardNameKey] as! String
+        let langName = info[kKeymanLanguageNameKey] as! String
+        let isRTL = (info[kKeymanKeyboardRTLKey] as! String) == "Y"
+        let font = info[kKeymanFontKey] as! String
+        let oskFont = info[kKeymanOskFontKey] as! String
+        KMManager.sharedInstance().addKeyboard(withID: kbID, languageID: langID, keyboardName: kbName,
+            languageName: langName, isRTL: isRTL, isCustom: false, font: font, oskFont: oskFont)
+        KMManager.sharedInstance().setKeyboardWithID(kbID, languageID: langID, keyboardName: kbName,
+            languageName: langName, font: font, oskFont: oskFont)
+      }
+    }
+
+    launchUrl = nil
+  }
+
+  func keyboardDownloadFailed(_ notification: Notification) {
+    if launchUrl != nil {
+      perform(#selector(self.dismissActivityIndicator), with: nil, afterDelay: 1.0)
+      let error = notification.userInfo?[NSUnderlyingErrorKey] as! Error
+      showAlert(withTitle: "Keyboard Download Error", message: error.localizedDescription,
+                cancelButtonTitle: "OK", otherButtonTitles: nil, tag: -1)
+      launchUrl = nil
+    }
+  }
+
+  func languagesUpdated(_ notification: Notification) {
+    updateStatus = 1
+  }
+
+  func languagesDownloadFailed(_ notification: Notification) {
+    updateStatus = -1
+  }
+
+  func keyboardPickerDismissed(_ notification: Notification) {
+    textView.becomeFirstResponder()
+    if UIDevice.current.userInterfaceIdiom == .pad {
+      perform(#selector(self.showGetStartedView), with: nil, afterDelay: 1.0)
+    }
+  }
+
```

Comment:
Hmm, `(void)showGetStartedView` has disappeared, though it's small enough that Swift may have optimized out its lone condition that would have prevented pass-through.

Reply:
Good catch. I was getting errors in disambiguating the two `showGetStartedView` in the `#selector` expressions so I removed that one function. It was only used in `keyboardPickerDismissed:` previously but I failed to add the check on whether the view should be shown.